macros/compare_which_columns_differ.sql

```diff
@@ -0,0 +1,46 @@
+{% macro compare_which_columns_differ(a_relation, b_relation, primary_key, exclude_columns=[]) %}
+    {{ return(adapter.dispatch('compare_which_columns_differ', 'audit_helper')(a_relation, b_relation, primary_key, exclude_columns)) }}
+{% endmacro %}
+
+{% macro default__compare_which_columns_differ(a_relation, b_relation, primary_key, exclude_columns=[]) %}  
+
+{% set column_names = dbt_utils.get_filtered_columns_in_relation(from=a_relation, except=exclude_columns) %}
+
+with bool_or as (
+
+    select 
+        true as anchor
+        {% for column in column_names %}
+            {% set column_name = adapter.quote(column) %}
+            {% set compare_statement %}
+                (a.{{ column_name }} != b.{{ column_name }}
+                or a.{{ column_name }} is null and b.{{ column_name }} is not null
+                or a.{{ column_name }} is not null and b.{{ column_name }} is null)
```

Comment:
Is this enough brackets? I think you might need
```suggestion
                ((a.{{ column_name }} != b.{{ column_name }})
                or (a.{{ column_name }} is null and b.{{ column_name }} is not null)
                or (a.{{ column_name }} is not null and b.{{ column_name }} is null))
```